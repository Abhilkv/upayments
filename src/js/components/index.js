import Button from './button/button';
import TextBar from './textBar/textBar';
import ProductCard from './productCard/productCard';
import Loader from './loader/loader';
import InputBox from './inputBox/inputBox';

export {
  Loader,
  Button,
  TextBar,
  ProductCard,
  InputBox
};
