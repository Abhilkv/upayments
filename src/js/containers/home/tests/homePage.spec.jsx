import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import HomePage from '../homePage';
import { Button } from '../../../components';

configure({ adapter: new Adapter() });

const mockClick = jest.fn();


describe('components/homePage', () => {
  it('Renders default options', () => {
    const component = shallow(
      <HomePage />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders with product click ', () => {
    const dummyData = [
      {
        message: 'Success',
        product: {
          _id: '62fc9dab69257f89cdb740c6',
          name: 'Cup',
          avatar: 'https://images.unsplash.com/photo-1577968897966-3d4325b36b61?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80',
          description: 'Elegant Cup',
          price: 14,
          category: 'Accessories',
          developerEmail: 'karaahmethkn@gmail.com',
          createdAt: '2022-08-17T07:50:03.232Z',
          updatedAt: '2022-08-17T07:50:03.232Z',
          __v: 0
        }
      }
    ];
    const component = shallow(
      <HomePage fetchDummyApi={mockClick} productsData={dummyData} />
    );
    component.find('.category').props().onClick();
    expect(mockClick).toHaveBeenCalled();
  });
});
