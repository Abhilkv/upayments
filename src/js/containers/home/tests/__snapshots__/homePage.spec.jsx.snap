// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/homePage Renders default options 1`] = `
<div
  className="productContainer"
>
  <div
    className="boxContainer"
  >
    <input
      className="searchBar"
      onChange={[Function]}
      placeholder="Apple Watch, Samsung S21, Macbook Pro iphone 11"
      value=""
    />
    <select
      className="category"
      id="category"
      name="category"
      onChange={[Function]}
      value=""
    >
      <option
        className="optionText"
        value=""
      >
        Category
      </option>
    </select>
  </div>
  <div
    className="fabButton"
  >
    <Link
      className="addButton"
      to="/create"
    >
      <span
        className="addCircle"
      />
    </Link>
  </div>
</div>
`;
